indexing-service/src/main/java/io/druid/indexing/worker/WorkerTaskMonitor.java

```diff
@@ -84,6 +84,19 @@ public WorkerTaskMonitor(
   public void start()
   {
     try {
+      // cleanup any old running task announcements which are invalid after restart
+      for (TaskAnnouncement announcement : workerCuratorCoordinator.getAnnouncements()){
```

Comment:
we probably need to wait for pathchildrencache to be initialized before we run this, otherwise we might not clean up everything.